lib/ace/layer/gutter.js

```diff
@@ -228,6 +191,154 @@ var Gutter = function(parentEl) {
+            }
+            if (row > lastRow)
+                break;
+
+            var cell = this.$createCell();
+            fragment.appendChild(this.$renderCell(cell, config, foldLine, row));
+
+            row++;
+        }
+        return fragment;
+    };
+    
+    this.$cacheCell = function(cell) {
+        this.element.removeChild(cell);
+        this.$cells.push(cell);
+    };
+    
+    this.$createCell = function() {
+        var cell = this.$cells.pop();
+        if (!cell) {
+            cell = dom.createElement("div");
+            cell.style.contain = "strict";
+    
+            cell.textNode = document.createTextNode('');
+            cell.appendChild(cell.textNode);
+            
+            cell.foldWidget = dom.createElement("span");
+            cell.appendChild(cell.foldWidget);
+        }
+        this.element.appendChild(cell);
+        return cell;
+    };
+    
+    this.$renderCell = function(element, config, fold, row) {
+        var session = this.session;
+        
+        var foldWidget = element.childNodes[1];
+        var textNode = element.childNodes[0];
+        
+        var firstLineNumber = session.$firstLineNumber;
+        
+        var session = this.session;
+        var breakpoints = session.$breakpoints;
+        var decorations = session.$decorations;
+        var gutterRenderer = session.gutterRenderer || this.$renderer;
+        var foldWidgets = this.$showFoldWidgets && session.foldWidgets;
+        var foldStart = fold ? fold.start.row : Infinity;
+        
+        var className = "ace_gutter-cell ";
+        if (breakpoints[row])
+            className += breakpoints[row];
+        if (decorations[row])
+            className += decorations[row];
+        if (this.$annotations[row])
+            className += this.$annotations[row].className;
+        if (element.className != className)
+            element.className = className;
+
+        var height = session.getRowLength(row) * config.lineHeight + "px";
+        if (height != element.style.height)
+            element.style.height = height;
+
+        if (foldWidgets) {
+            var c = foldWidgets[row];
+            // check if cached value is invalidated and we need to recompute
+            if (c == null)
+                c = foldWidgets[row] = session.getFoldWidget(row);
+        }
+
+        if (c) {
+            var className = "ace_fold-widget ace_" + c;
+            if (c == "start" && row == foldStart && row < fold.end.row)
+                className += " ace_closed";
+            else
+                className += " ace_open";
+            if (foldWidget.className != className)
+                foldWidget.className = className;
+
+            var height = config.lineHeight + "px";
+            if (foldWidget.style.height != height)
+                foldWidget.style.height = height;
+                
+            if (foldWidget.style.display != "block")
+                foldWidget.style.display = "block";
```

Comment:
this needs to be `inline-block`, to not go to the next line, so may be better to use `.display =  ""` instead

Reply:
changed